src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+          this.visitedWizardlets.push(current);
+        }
+        return current;
+      }
+    },
+    {
+      name: 'currentSection',
+      expression: function(currentWizardlet, wizardPosition$sectionIndex) {
+        return this.currentWizardlet.currentSection = currentWizardlet
+            .sections[wizardPosition$sectionIndex];
+      }
+    },
+    {
+      name: 'isLastScreen',
+      class: 'Boolean',
+      expression: function(nextScreen) {
+        return nextScreen == null;
+      }
+    },
+    {
+      name: 'canGoBack',
+      class: 'Boolean',
+      expression: function(previousScreen) {
+        return previousScreen != null;
+      }
+    },
+    {
+      name: 'canGoNext',
+      class: 'Boolean',
+      expression: function(
+        wizardPosition, nextScreen,
+        currentSection, currentWizardlet,
+        currentSection$isValid, currentWizardlet$isValid,
+        config$allowSkipping
+      ) {
+        if ( config$allowSkipping ) return true;
+        if (
+          ! nextScreen ||
+          wizardPosition.wizardletIndex != nextScreen.wizardletIndex
+        ) {
+          if ( ! currentWizardlet$isValid ) return false;
+        }
+        return currentSection$isValid;
+      }
+    },
+    {
+      name: 'submitted',
+      class: 'Boolean'
+    }
+  ],
+
+  methods: [
+    function setupWizardletListeners(wizardlets) {
+      console.log('step wizard', this, wizardlets);
+      if ( this.wsub ) this.wsub.detach();
+      this.wsub = this.FObject.create();
+
+      var wi = 0, si = 0;
+      wizardlets.forEach((w, wizardletIndex) => {
+
+        // Bind availability listener for wizardlet availability
+        var isAvailable$ = w.isAvailable$;
+        console.log(wizardletIndex, w, w.isAvailable$, isAvailable$);
+        this.wsub.onDetach(isAvailable$.sub(() => {
+          console.log('X');
+          this.onWizardletAvailability(wizardletIndex, isAvailable$.get());
+        }));
+
+        // Bind availability listener for each wizardlet section
+        w.sections.forEach((section, sectionIndex) => {
+          var sectionPosition = this.WizardPosition.create({
+            wizardletIndex: wizardletIndex,
+            sectionIndex: sectionIndex,
+          });
+          var isAvailable$ = section.isAvailable$;
+          this.wsub.onDetach(isAvailable$.sub(() => {
+            console.log('Y');
+            this.onWizardPositionAvailabilityUpdate(
+              sectionPosition, isAvailable$.get());
+          }));
+        });
+
+      });
+
+    },
+    function detach() {
+      this.wsub.detach();
+      this.SUPER();
+    },
+    function positionAfter(pos) {
+      let subWi = pos.wizardletIndex
+      let subSi = pos.sectionIndex;
+      if ( subSi >= this.wizardlets[subWi].sections.length - 1 ) {
+        if ( subWi >= this.wizardlets.length - 1 ) return null;
+        subSi = 0;
+        subWi++;
+      } else {
+        subSi++;
+      }
+      return this.WizardPosition.create({
```

Comment:
Why create a new object? Wouldn't reassigning the properties be enough?

Reply:
WizardPosition? We've been using that for a while now. It's simplified a lot of the wizard by making the wizard position a single slot you can listen two rather than two slots that can change independently.